cmd/hooks/prestop.go

```diff
@@ -0,0 +1,113 @@
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+func PreStop(clientset kubernetes.Interface, timeout time.Duration) error {
+	klog.InfoS("PreStop: executing PreStop lifecycle hook")
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	nodeName := os.Getenv("CSI_NODE_NAME")
+	if nodeName == "" {
+		return fmt.Errorf("PreStop: CSI_NODE_NAME missing")
+	}
+
+	node, err := fetchNodeInfo(ctx, clientset, nodeName)
+	if err != nil {
+		return err
+	}
+
+	if isNodeBeingDrained(node) {
+		klog.InfoS("PreStop: node is being drained, checking for remaining VolumeAttachments", "node", nodeName)
+		return waitForVolumeAttachments(ctx, clientset, nodeName)
+	}
+
+	klog.InfoS("PreStop: node is not being drained, skipping VolumeAttachments check", "node", nodeName)
+	return nil
+}
+
+func fetchNodeInfo(ctx context.Context, clientset kubernetes.Interface, nodeName string) (*v1.Node, error) {
```

Comment:
Suggest renaming this to `fetchNode`, as it is fetching the entire node (object), not "node info"

Reply:
Also, not in scope for this PR, but I wonder if it's worth creating a "k8s utils" style package that does this kind of common operation? We probably have nearly identical code to this in the taint removal feature.

Reply:
> I wonder if it's worth creating a "k8s utils" style package that does this kind of common operation

This is an excellent idea 👍 

> Suggest renaming this to fetchNode

Done.